tests/mock_network/mod.rs

```diff
@@ -464,3 +464,22 @@ fn check_section_info_ack() {
     let expected_all: Vec<_> = nodes.iter().map(|node| node.id()).collect();
     assert_eq!(node_with_sibling_knowledge, expected_all);
 }
+
+#[test]
+fn carry_out_parsec_pruning() {
+    let network = Network::new(MIN_SECTION_SIZE, None);
+    let mut nodes = create_connected_nodes(&network, 2 * MIN_SECTION_SIZE);
+
+    for node in nodes.iter_mut() {
+        let _ = node
+            .inner
+            .node_state_mut()
+            .map(|state| state.vote_for_prune());
+    }
+    poll_and_resend(&mut nodes, &mut []);
+
+    nodes = create_connected_nodes_until_split(&network, vec![1, 1], true);
+    poll_and_resend(&mut nodes, &mut []);
+
+    verify_invariant_for_all_nodes(&network, &mut nodes);
```

Comment:
That is pretty good, Ideally we would also test that prunning catually happen (i.e it seem this test would pass even if prunning does nothing.

Reply:
agreed, I will expand the test